### DISCUSSION TITLE
This looks awesome

### DISCUSSION THREAD

**Discussion by ryantheleach** [mizarc/bell-claims#96] [General]
This looks like the GUI interface I always dreamed that Grief Protection would have.

I used to be paranoid about GUI menu's causing dupes, but it seems like the community has worked out mature ways of making menu's now.

**Comment by mizarc** [comment]
Thanks! I actually started this plugin because I didn't find the popular GriefPrevention plugin intuitive enough for my user base. GUIs was the primary focus so I had to make sure everything worked flawlessly.

Hope you continue to enjoy what I'm building here, and feel free to share it around!

**Reply by ryantheleach** [reply]
Not sure if the Bukkit/Spigot GP changed hands, but I know a while back that the sponge fork got a little harder to use because of all the mods it had to support and excess flags, and that the bukkit page had it's docs completely rewritten for the worse to cover new features.

I haven't had the pleasure of trying it yet so don't know if you do this, but I highly recommend warning the player on first bed / chest like GP used to do, as that was 100% the best entrypoint for players to learn the system.

Setting a spawn / creating a chest is kinda the first signs of a home.

I like that it uses Bells to manage it, as it's a rarer block that can serve as a center.

Do you add recipes for Bells? or are admins expected to give them out, or players find them in villages?

**Reply by mizarc** [reply]
There's currently no system in place to let people know how to claim when they place a bed/chest. I can definitely take it into consideration just to let them know.

Currently I rely on the server administrators to figure out how best to let players know about the system and how to use it. On my own server as well as many other servers I've seen, a custom trade villager is set up at spawn to allow players to buy bells with signs explaining how to use the system. Adding a crafting recipe is also a cool option that can be done, it really feels like a bell recipe should be a vanilla mechanic but ah well.

I can always add a custom recipe option into the plugin, though it will depend on what would be best. Gold seems like the obvious choice to put in the recipe, though most players do expect to be able to claim from the get-go so expecting them to get gold to make their first claim seems far-fetched.

**Reply by ryantheleach** [reply]
I haven't started my server yet (It comes and goes every 3 years or so, hobbies be like that) But one thing I was considering was keeping track of the first chest placed by a player, and just, placing a bell inside as soon as it's placed. It seemed, weird, but it was what my monkey-patch was going to be if I ended up using bell-claims.

That way I could let players have 1 claim for free, then monitor and poll the players about opinions on the best way to handle further plots going in.

The other option I've used in the past, is first-join kits, that have heavily customized items with lore explaining the various systems in place.

E.g. giving them a bed, that explains how timeskip works, giving them a golden shovel, with instructions on grief protection, but some players found the information overload + free items disconcerting, as they wanted to enjoy the early vanilla survival experience without free stuff.

**Reply by mizarc** [reply]
Those are for sure creative solutions to the bell obtaining problem. Each server does like to have their own way to let players claim, whether it be a free handout or something you have to earn. I always like to see what kind of solutions server administrators have, so I'm not sure if I need to have an official way to make obtaining bells easier for the core plugin itself. There are definitely other plugins out there you can use to modify recipes/kits/trades/etc.
